### ISSUE TITLE
Each Jenkins build adds a new row 'Report title' inside HTML Classifications table

### ISSUE THREAD

**Issue by fslev** [jenkinsci/cucumber-reports-plugin#359]
Configure the Cucumber reports section for a Jenkins Job, by setting 'Report title': 'Nightly'.

![Screenshot from 2020-05-28 11-39-07](https://user-images.githubusercontent.com/17047317/83119058-e8100c00-a0d7-11ea-902c-bdc32ae9c263.png)

Run the job twice.

Actual:
Inside the generated Cucumber reports HTML page, the classifications table contains 'Report title' twice.
Also, inside Jenkins Job configuration -> Cucumber reports -> Classifications , 'Report title' is automatically generated and duplicated after each job run:

![Screenshot from 2020-05-28 11-37-30](https://user-images.githubusercontent.com/17047317/83119234-21487c00-a0d8-11ea-92df-8bb8441a22a8.png)

**Comment by damianszczepanik** [comment]
If your configuration is changed after job running I believe you should review your build configuration as running should not update the job configuration at all

**Comment by fslev** [comment]
I know.It looks fanatstic. But it seems that is happening. After each job run, a new classification is generated inside job configuration. There is nothing to review inside my job configuration, since it's very simple and I am using corectly all Cucumber reports fields.
Ok, we'll leave the issue closed. I will deactivate the Report Title feature.Though I find it hard to believe this issue is on my side, since I've repoduced it on two different Jenkins instances.

**Comment by damianszczepanik** [comment]
Then share the configuration

**Comment by damianszczepanik** [comment]
But before that switch on the audit https://plugins.jenkins.io/jobConfigHistory/ and verify

**Comment by fslev** [comment]
Attached job configuration.
[configOutput.txt](https://github.com/jenkinsci/cucumber-reports-plugin/files/4767481/configOutput.txt)
It seems there are no differences in terms of configuration history between consecutive builds.
Also, it seems the Cucumber Reports classification properties do not appear at all inside that configuration.
I believe it has something to do with caching...maybe

**Comment by damianszczepanik** [comment]
So maybe I don't understand the issue, what do you mean by 'new classification'.
Based on your config, classification is empty.

**Comment by fslev** [comment]
By 'new classification', I mean duplicated classification property (name and value), inside 'Presentation' section of the Cucumber report Jenkins plugin. This classification property actually represents the 'report title' and is duplicated after each build.

**Comment by damianszczepanik** [comment]
I was not able to reproduce it, can you help about it?

**Comment by fslev** [comment]
This should be easily be reproduced, just by setting a value for 'Report Title' inside Jenkins and generate two or three builds.  Also, by looking inside the code, it seems quite logic where the problem is.
Maybe different Jenkins versions use this plugin differently ?

**Comment by damianszczepanik** [comment]
If this would be so easy then the bug will be reported, confirmed and fixed year or two ago as this plugin is quite popular :)

Also it is not clear to me how running job can change its configuration. It happened to me when using pipeline approach but it wasn't persistent change.

**Comment by fslev** [comment]
This bug is strictly related to the publish multiple reports feature, which was added on 17th May:

https://github.com/jenkinsci/cucumber-reports-plugin/commit/71c15e2f3a2e584ce29a399cd71de25cf334ee45

Since then, two issues were reported:
The current one, and #365

**Comment by fslev** [comment]
> Also it is not clear to me how running job can change its configuration. It happened to me when using pipeline approach but it wasn't persistent change.
Running the job can change the configuration of the plugin as follows:
-> every time the plugin is invoked after the build, report title is dynamically added inside the 'classifications' list from CucumberReportPublisher class:
```
 if (StringUtils.isNotEmpty(reportTitle)) {
            classifications.add(new Classification(Messages.Classification_ReportTitle(), reportTitle));
        }
```
The 'classifications' are then displayed inside the plugin configuration via config.jelly from src/main/resources/net/masterthought/jenkins/CucumberReportPublisher:
```
<f:entry
                    field="classifications"
                    title="${%classifications.title}">
                <f:repeatableProperty field="classifications"/>
            </f:entry>
```

**Comment by damianszczepanik** [comment]
Fixed in #371 please review and test before releasing
